src/functional/functional.h

```diff
@@ -705,6 +705,56 @@ class FunctionalErrorNormLpBoundary : public Functional<dim,nstate,real,MeshType
     const bool                use_all_boundaries;
 };
 
+///Functional to take the integral of the solution
+#if PHILIP_DIM==1
+    template <int dim, int nstate, typename real, typename MeshType = dealii::Triangulation<dim>>
+#else
+    template <int dim, int nstate, typename real, typename MeshType = dealii::parallel::distributed::Triangulation<dim>>
+#endif
+class SolutionIntegral : public Functional<dim,nstate,real,MeshType>
+{
+public:
+    using FadType = Sacado::Fad::DFad<real>; ///< Sacado AD type for first derivatives.
+    using FadFadType = Sacado::Fad::DFad<FadType>; ///< Sacado AD type that allows 2nd derivatives.
+public:
+    /// Constructor
+    SolutionIntegral(
+            std::shared_ptr<PHiLiP::DGBase<dim,real, MeshType>> dg_input,
+            std::shared_ptr<PHiLiP::Physics::PhysicsBase<dim,nstate,FadFadType>> _physics_fad_fad,
+            const bool uses_solution_values = true,
+            const bool uses_solution_gradient = false)
+            : PHiLiP::Functional<dim,nstate,real,MeshType>(dg_input,_physics_fad_fad,uses_solution_values,uses_solution_gradient)
+    {}
+
+    /// Templated volume integrand
+    template <typename real2>
+    real2 evaluate_volume_integrand(
+            const PHiLiP::Physics::PhysicsBase<dim,nstate,real2> &physics,
+            const dealii::Point<dim,real2> &phys_coord,
+            const std::array<real2,nstate> &soln_at_q,
+            const std::array<dealii::Tensor<1,dim,real2>,nstate> &soln_grad_at_q) const;
+
+    /// Non-template functions to override the template classes
+    real evaluate_volume_integrand(
+            const PHiLiP::Physics::PhysicsBase<dim,nstate,real> &physics,
+            const dealii::Point<dim,real> &phys_coord,
+            const std::array<real,nstate> &soln_at_q,
+            const std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_at_q) const override
+    {
+        return evaluate_volume_integrand<>(physics, phys_coord, soln_at_q, soln_grad_at_q);
```

Comment:
```suggestion
        return evaluate_volume_integrand<real>(physics, phys_coord, soln_at_q, soln_grad_at_q);
```

Reply:
This actually doesn't compile. You'll notice that all other functionals in functional.h have empty brackets. I believe it will deduce the right argument. Let's leave it as it is.